otter/grade/containers.py

```diff
@@ -117,9 +117,28 @@ def launch_containers(
 
     # stop execution while containers are running
     finished_futures = wait(futures)
-
-    # return list of dataframes
-    return [df.result() for df in finished_futures[0]]
+    
+    # return list of dataframes with points possible as first frame
+    full_df = []
+    f = next(iter(finished_futures[0]))
+    scores = f.result()
+    scores_dict = scores.to_dict()
+
+    pts_poss_dict = {t: [scores_dict[t]["possible"]] for t in scores_dict}
+    pts_poss_dict[KEY_SCORES_DICT_FILE] = POINTS_POSSIBLE_LABEL
+    pts_poss_dict[KEY_SCORES_DICT_PCT_CRT] = "NA"
+    pts_poss_dict[KEY_SCORES_DICT_TTL_PTS] = scores.possible
+    pts_poss_df = pd.DataFrame(pts_poss_dict)
+    full_df.append(pts_poss_df)
+    for f in finished_futures[0]:
+        scores_dict = f.result().to_dict()
+        scores_dict = {t: [scores_dict[t]["score"]] for t in scores_dict}
+        scores_dict[KEY_SCORES_DICT_PCT_CRT] = round(f.result().total / f.result().possible,4)
+        scores_dict[KEY_SCORES_DICT_TTL_PTS] = f.result().total
+        scores_dict[KEY_SCORES_DICT_FILE] = f.result().file
+        df_scores = pd.DataFrame(scores_dict)
+        full_df.append(df_scores)
```

Comment:
Can you put this logic for converting a `list[GradingResults]` to a dataframe into a helper function in otter/grade/utils.py and use it here? Having all of this logic here makes the code hard to follow.

```python
def merge_scores_to_df(scores: List[GradingResults]) -> pd.DataFrame:
    """
    Convert a list of ``GradingResults`` objects to a scores dataframe, including a row
    with the total point values for each question.

    Args:
        scores (``list[otter.test_files.GradingResults]``): the score objects to merge

    Returns:
        ``pd.DataFrame``: the scores dataframe
    """
    ...
```